proto/commonfate/access/v1alpha1/access.proto

```diff
@@ -283,6 +283,15 @@ message BatchEnsureResponse {
   //
   // Rather than failing the entire operation the Access Handler returns warnings.
   repeated Diagnostic diagnostics = 6;
+
+  DurationConfiguration duration_configuration = 7;
```

Comment:
Im wondering if this belongs on the Grant of on the batch ensure.
I know in the frontend that we only show the lowest max duration value on the single duration picker.

But it feels like maybe we shouldn't bake that into the API.

I think it's reasonable to put it on each grant, because when you use batchEnsure, you provide a duration for each grant individually.

Reply:
I was thinking the opposite, the grant should represent a snapshot of a request of an entitlement. Which only holds information about the grant as it is when it was made. I was thinking that storing the duration information on the grant was an incorrect way of transporting this information to the client. This is actually how it was previously (although there was only duration and max duration) and handling the different durations on the grant was confusing. Unless we separate the final duaration with the configuration duration fields on the grant eg. 

```
grant {
   duration: 3600

   durationConfig {
       maxDuration: 6000
       defaultDuration: 3600
   }
}
```

> because when you use batchEnsure, you provide a duration for each grant individually.
At the moment this is not the case and the duration is for the request rather than each grant.

Reply:
Maybe it can go on the GrantState which is specific to the batchEnsure 
```diff
message GrantState {
  Grant grant = 1;
  // The change to the grant.
  // Will be unspecified if the grant has not been changed.
  GrantChange change = 2;
+ // will be provided when the grant is going to be created
+ optional DurationConstraints duration_constraints = 3;
}
```

Reply:
But then again, maybe we just go with your approach for now, where the server gives back the max conditions for all the grants requested. and we leave it as is.

It can be changed later if we need to